lib/streamlit/auth_util.py

```diff
@@ -144,6 +173,177 @@ def generate_default_provider_section(auth_section: AttrDict) -> dict[str, Any]:
     return default_provider_section
 
 
+def set_cookie_with_chunks(
+    set_single_cookie_fn: Callable[[str, str], None],
+    create_signed_value_fn: Callable[[str, str], bytes],
+    cookie_name: str,
+    value: dict[str, Any],
+) -> None:
+    """Set a cookie, splitting into multiple cookies if necessary.
+
+    Args:
+        set_single_cookie_fn: Function to set a single cookie (cookie_name, value)
+        create_signed_value_fn: Function to create a signed cookie value (cookie_name, value)
+        cookie_name: Name of the cookie
+        value: Dictionary value to serialize and store
+    """
+    serialized_cookie_value = json.dumps(value)
+
+    # Calculate actual cookie size using the provided signing function
+    signed_value = create_signed_value_fn(cookie_name, serialized_cookie_value)
+
+    # Cookie format: "name=value; HttpOnly; Path=/"
+    # Approximate additional bytes for cookie attributes (~30 bytes)
+    actual_cookie_size = len(cookie_name) + 1 + len(signed_value) + 30
```

Comment:
What is the +1 here?

I guess constants would be nice here instead of the comment
`len(cookie_name) + SOME_CONSTANT + len(signed_value) + APPROX_COOKIE_ATTR_SIZE`